### ISSUE TITLE
Permissions are incorrectly set before ZIP/Upload

### ISSUE THREAD

**Issue by matthewdfuller** [XT-i/aws-lambda-jenkins-plugin#12]
When using the Lambda plugin, I'm getting an error executing the function in Lambda:

```
EACCES, permission denied '/var/task/index.js

{
  "errorMessage": "EACCES, permission denied '/var/task/index.js'",
  "errorType": "Error",
  "stackTrace": [
    "Object.fs.openSync (fs.js:439:18)",
    "Object.fs.readFileSync (fs.js:290:15)",
    "Object.Module._extensions..js (module.js:473:44)",
    "Module.load (module.js:356:32)",
    "Function.Module._load (module.js:312:12)",
    "Module.require (module.js:364:17)",
    "require (module.js:380:17)"
  ]
}
```

On further investigation, it seems to be a permissions issue with Jenkins. I'm not entirely sure if this is directly related to the plugin (so feel free to mark as invalid), but when I added a manual step that "chmod -R" with more open permissions before the Lambda function executed, it worked.

I also filed a bug with AWS for this, so perhaps it is Lambda's issue, but it seems that the permissions on the files are lost pre-zipping.

**Comment by cast** [comment]
Hi Matthew, I'm fairly positive that the Amazon AWS service is not to blame, so in my opinion you could close that ticket. Could you check the file permissions inside the workspace and if no apparent issue is found, don't hesitate to give me more information about the system you use as build server..

The plugin first checks if the given location is a file, folder or s3 location and if necessary creates and copies over the zip file to a temp file to prepare for upload.

Thanks,

Michael

**Comment by matthewdfuller** [comment]
Hey Michael, Thanks for the response - I closed the ticket with AWS (you were right, they just said the file permissions were wrong).

I tried adding a build step before Lambda is called that just runs chmod -R 644 on the whole workspace. This seemed to fix the problem, so it is leading me to believe this is an issue on my build system and not with your plugin (which works very well once I fixed the permissions!).

Apologies for not investigating this more before opening the issue, I'm going to close it for now.

Matt
